cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -310,6 +338,16 @@ func (s *Server) pullChartTarball(ctx context.Context, packageRef *corev1.Availa
 	return waitUntilChartPullComplete(watcher)
 }
 
+// TODO:
+// https://github.com/kubeapps/kubeapps/pull/2915
+// @minelson says Nice - great use of both channel and the watcher to handle the
+// async request. Note that this effectively makes pullChartTarball a blocking call,
+// which is fine in it's current use (where it's only called in a context of one chart).
+// Not for now, but in the future you might instead want to consider something like
+// passing a results channel (of string urls) to pullChartTarball, so it returns
+// immediately and you wait on the results channel at the call-site, which would mean
+// you could call it for 20 different charts and just wait for the results to come in
+//  whatever order they happen to take, rather than serially.
```

Comment:
Same here. For example, state the TODO and link to more context (the gh issue comment).

Reply:
ok